pkg/providers/instancetype/instancetype.go

```diff
@@ -77,9 +77,26 @@ func (p *Provider) CreateOfferings(shape *internalmodel.WrapShape, zones sets.Se
 	if p.priceSyncer != nil {
 		priceCatalog = &p.priceSyncer.PriceCatalog
 	}
+	basePrice := float64(pricing.Calculate(shape, priceCatalog))
 
 	// only on-demand support
 	for zone := range zones {
+		// Add preemptible offering with lower price (higher priority)
```

Comment:
Could you help to add some unit test for preemptible instance provision in file like pkg/cloudprovider/suite_test.go